src/warnet/bitcoin.py

```diff
@@ -35,15 +40,37 @@ def rpc(tank: str, method: str, params: str):
     print(result)
 
 
-def _rpc(tank: str, method: str, params: str):
+def _rpc(tank: str, method: str, params: tuple[str, ...]) -> str:
     # bitcoin-cli should be able to read bitcoin.conf inside the container
     # so no extra args like port, chain, username or password are needed
     namespace = get_default_namespace()
+
+    sclient = get_static_client()
     if params:
-        cmd = f"kubectl -n {namespace} exec {tank} -- bitcoin-cli {method} {' '.join(map(str, params))}"
+        cmd = ["bitcoin-cli", method]
+        cmd.extend(params)
+    else:
+        cmd = ["bitcoin-cli", method]
+    resp = stream(
+        sclient.connect_get_namespaced_pod_exec,
+        tank,
+        namespace,
+        container="bitcoincore",
+        command=cmd,
+        stderr=True,
+        stdin=False,
+        stdout=True,
+        tty=False,
+    )
+    # The k8s lib seems to convert json into its python representation. The following dance seems to
+    # avoid the worst of it.
+    if resp.startswith("{") or resp.startswith("["):
+        literal = ast.literal_eval(resp)
+        json_string = json.dumps(literal)
+        return json_string
```

Comment:
the old `exec_run()` function added this option to `stream()` for this: ` _preload_content=False`

overall i think its better to restore the old `exec_run()` function in k8s.py from pre-italy and just call it here.
That's as far as I got on this project: 756df48b97acb1496b2958e12b895a00

Reply:
addressed here via _preload_content=False: https://github.com/bitcoin-dev-project/warnet/pull/614/commits/16f869a2a3725ae61f895e153320d6ea18a13add